api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -226,8 +211,33 @@ public void setNextSequenceValue(Long nextSequenceValue) {
 		this.nextSequenceValue = nextSequenceValue;
 	}
 	
-	private PrefixProvider getPrefixProvider(String prefixProviderBean) {
-		return Context.getRegisteredComponent(prefixProviderBean, PrefixProvider.class);
+	/**
+	 * Gets the {@link PrefixProvider} from a configured {@code prefix}
+	 * 
+	 * @should return null if a static prefix is provided
+	 * @should return null if the configured provider is not found
+	 * @param prefix
+	 * @return the PrefixProvider
+	 */
+	public PrefixProvider getPrefixProvider(String prefix) {
+		if (StringUtils.isBlank(prefix)) {
+			return null;
+		}
+		if (prefix.startsWith(CONFIGURATION_PREFIX)) {
+			String prefixProviderBean = "";
+			try {
+				prefixProviderBean = prefix.substring(prefix.indexOf(":") + 1).trim();
+				return Context.getRegisteredComponent(prefixProviderBean, PrefixProvider.class);
+			}
+			catch (IndexOutOfBoundsException e) {
+				log.error("Invalid Prefix Configuration. PrefixProvider bean name not provided", e);
+			}
+			catch (APIException e) {
+				log.error("Failed to retrieve PrefixProvider:" + prefixProviderBean, e);
+			}
+		}
+		return null;
```

Comment:
Instead of a last resort `null` value you could also return a `new ConstantPrefixProvider(prefix)` so that in the end you could always make the call eventually:
```java
prefixProvider.getValue()
```
In this way we would always treat the member `prefix` as prefix _configuration_, the default configuration being a constant prefix one:
```java
import org.apache.commons.lang.StringUtils;

public class ConstantPrefixProvider implements PrefixProvider {

	private String prefix = "";
	
	public ConstantPrefixProvider(String prefix) {
		if (!StringUtils.isEmpty(prefix)) {
			this.prefix = prefix;
		}
	}
	
	@Override
	public String getValue() {
		return prefix;
	}
}
```